pylops_mpi/DistributedArray.py

```diff
@@ -49,14 +50,45 @@ def local_split(global_shape: Tuple, base_comm: MPI.Comm,
         local_shape = global_shape
     # Split the array
     else:
+        # TODO: Both NCCL and MPI have to use MPI.COMM_WORLD for shape splitting
+        # So it was decided to make it explicit. This will make base_comm unused.
         local_shape = list(global_shape)
-        if base_comm.Get_rank() < (global_shape[axis] % base_comm.Get_size()):
-            local_shape[axis] = global_shape[axis] // base_comm.Get_size() + 1
+        if MPI.COMM_WORLD.Get_rank() < (global_shape[axis] % MPI.COMM_WORLD.Get_size()):
+            local_shape[axis] = global_shape[axis] // MPI.COMM_WORLD.Get_size() + 1
         else:
-            local_shape[axis] = global_shape[axis] // base_comm.Get_size()
+            local_shape[axis] = global_shape[axis] // MPI.COMM_WORLD.Get_size()
     return tuple(local_shape)
 
 
+def subcomm_split(mask, base_comm: MPI.Comm = MPI.COMM_WORLD):
+    """Create new communicators based on mask
+    This method creates new NCCL communicators based on ``mask``.
+    Contrary to MPI, NCCL does not provide support for splitting of a communicator in multiple subcommunicators;
+    this is therefore handled explicitly by this method.
+
+    Parameters
+    ----------
+    mask : :obj:`list`
+        Mask defining subsets of ranks to consider when performing 'global'
+        operations on the distributed array such as dot product or norm.
+
+    base_comm : :obj:`mpi4py.MPI.Comm`, optional
+        A Communicator over which array is distributed
+        Defaults to ``mpi4py.MPI.COMM_WORLD``.
+
+    Returns:
+    -------
+        Union[mpi4py.MPI.Comm, cupy.cuda.nccl.NcclCommunicator]]: a subcommunicator according to mask
```

Comment:
see comment in other places about conforming to the doctoring style of pylops/pylops-mpi... this should be

```
sub_comm : :obj:`mpi4py.MPI.Comm` or :obj:`cupy.cuda.nccl.NcclCommunicator`
        Subcommunicator according to mask
```